tower/tests/buffer/main.rs

```diff
@@ -301,6 +316,14 @@ async fn wakes_pending_waiters_on_failure() {
     assert_pending!(worker.poll());
     let mut response = task::spawn(service1.call("hello"));
 
+    assert!(worker.is_woken(), "worker task should be woken by request");
+    assert_pending!(worker.poll());
+
+    // fill the channel so all subsequent requests will wait for capacity
+    let service1 = assert_ready_ok!(task::spawn(service.ready()).poll());
+    assert_pending!(worker.poll());
+    let mut response2 = task::spawn(service1.call("world"));
```

Comment:
this test required adding an additional message to the channel for the buffer to be "full", I described this in the PR description